washingtonpost/FastFEC
# [DENG-129] Catch "Unmatched for version" error as python exception

## Description
Simply `exit(0)`ing upon an unmatched version made it such that I couldn't simply catch a filing that failed to parse and move on. This PR causes the `unmatched version` error to `return -1` up through its stack until I can raise an exception in Python.

This seems to work in principle, but I have the following concerns:
* Is raising just an `Exception` sufficient since the actual error is being printed to `stderr` beforehand?
* Is there anything I need to do re: cleaning up memory? I raise the exception in `client.py` after the `# Free memory and file descriptors`, but unsure if I need to do anything else.
* I welcome any other suggestions!!


## Test Steps
I also made a little test script to make sure this gets caught.
```
from fastfec import FastFEC
import smart_open
import logging

if __name__ == "__main__":
    filing_id = 1613679 # broken
    # filing_id = 1613811 # works
    
    # import os
    # _TEST_FOLDER = os.path.dirname(__file__)
    # filing_id=1613679
    # fec_source_url = os.path.join(_TEST_FOLDER, "1613679_malformed.fec")
    fec_source_url = f"http://docquery.fec.gov/dcdev/posted/{filing_id}.fec"


    with smart_open.open(
        fec_source_url,
        "rb",
        transport_params={"headers": {"User-Agent": "Mozilla/5.0"}},
    ) as f:
        with FastFEC() as fastfec:
            try:
                logging.info(
                    f"Extracting from {fec_source_url} to s3 for filing_id {filing_id}"
                )
                fastfec.parse_as_files(
                    f, 'output/', include_filing_id=str(filing_id)
                )

            except Exception as e:
                logging.error(f"Failed to parse filing_id {filing_id}. Skipping. {e}")
```
which returns
```
FastFEC % python python/src/fastfec/test.py
Error: Unmatched for version 8.4 and form type F3Z1
ERROR:root:Failed to parse filing_id 1613679. Skipping. FastFEC failed to parse.
```

## After Screenshot(s)

## Before Screenshot(s)
